src/webui/api/rsscontroller.cpp

```diff
@@ -91,7 +94,28 @@ void RSSController::itemsAction()
     setResult(jsonVal.toObject());
 }
 
-void RSSController::refreshItemAction()
+void RSSController::markAsReadAction() const
+{
+    requireParams({"itemPath"});
+
+    const QString itemPath {params()["itemPath"]};
+    const QString articleId {params()["articleId"]};
+
+    RSS::Item *item = RSS::Session::instance()->itemByPath(itemPath);
+
+    if (!item) return;
```

Comment:
Although RSS WebAPI is currently simplified and does not have any error reporting, it would be nice to implement it in the future. Client should know if something goes wrong.